django/core/cache/backends/db.py

```diff
@@ -46,41 +45,44 @@ class CacheEntry(object):
 class DatabaseCache(BaseDatabaseCache):
 
     # This class uses cursors provided by the database connection. This means
-    # it reads expiration values as aware or naive datetimes depending on the
-    # value of USE_TZ. They must be compared to aware or naive representations
-    # of "now" respectively.
+    # it reads expiration values as aware or naive datetimes, depending on the
+    # value of USE_TZ and whether the database supports time zones. They must
+    # be compared to aware or naive representations of "now" respectively.
 
     # But it bypasses the ORM for write operations. As a consequence, aware
     # datetimes aren't made naive for databases that don't support time zones.
-    # We work around this problem by always using naive datetimes when writing
-    # expiration values, in UTC when USE_TZ = True and in local time otherwise.
+
+    # We work around this by calling directly the adapters or converters.
 
     def get(self, key, default=None, version=None):
         key = self.make_key(key, version=version)
         self.validate_key(key)
         db = router.db_for_read(self.cache_model_class)
-        table = connections[db].ops.quote_name(self._table)
+        connection = connections[db]
+        table = connection.ops.quote_name(self._table)
 
-        with connections[db].cursor() as cursor:
+        with connection.cursor() as cursor:
             cursor.execute("SELECT cache_key, value, expires FROM %s "
                            "WHERE cache_key = %%s" % table, [key])
             row = cursor.fetchone()
         if row is None:
             return default
-        now = timezone.now()
         expires = row[2]
-        if connections[db].features.needs_datetime_string_cast and not isinstance(expires, datetime):
-            # Note: typecasting is needed by some 3rd party database backends.
-            # All core backends work without typecasting, so be careful about
-            # changes here - test suite will NOT pick regressions here.
-            expires = typecast_timestamp(str(expires))
-        if expires < now:
+
+        if hasattr(connection.ops, 'convert_datetimefield_value'):
+            # XXX This relies on convert_datetimefield_value not using its
```

Comment:
I would like to address this further. The only problem is that I was unable to figure out what the parameters are supposed to be.

@mjtamlyn @jarshwah @akaariai Perhaps one of you can help?

It would be nice to make it easier for future contributors to grok this API.